### PR TITLE
feat(fadec): Impl FADEC GND PWR buttons + corrected Igniter logic

### PR DESCRIPTION
**Pull request by Maximilian-Reuter**
Fixes #[issue_no]

## Summary of Changes
This PR implements the FADEC GND PWR pushbuttons for each FADEC. It also implements the autoswitching logic 
of igniters on an automatic groundstart. Previously it would always display both igniters in use on startup.
Now it only uses one igniter and alternates after every startup attempt.

Crankmode now also energizes the FADEC.

I also moved the igniter code into the A32NX_FADEC.js file since it seemed more appropriate. 

## References
Since I didn't have access to NEO documentation, most of the logic here is based on videos and publicly available CEO documentation. Should any discrepancy be discovered I'd be happy to adjust my code.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

The test regiment should manly deal with FADEC related logic.

1. Spawn C/D
2. Power Up A/C using ground power
3. Wait 5 minutes for initial FADEC power up period to elapse and EICAS to show amber X
4. Use the FADEC GND PWR 1+2 Pushbuttons to activate the respective Fadecs and watch the EICAS show values
5. Turn off pushbuttons and see amber X on EICAS
6. Move Engine Mode Selector into CRANK 
7. Watch EICAS display values instead of amber X
8. Move Engine Mode Selector to NORM 
9. Watch EICAS display amber X
10. Move Engine Mode Selector to IGN/START
11. Watch EICAS display values instead of amber X
12. Move Engine Mode Selector to NORM 
13. Wait 30 seconds
14. Watch EICAS display amber X
15. Turn on APU and APU BLEED and turn on Fuel Pumps
16. Move Engine Mode Selector to IGN/START
17. Move Engine Master SW to ON
18. Watch EICAS until N2 reaches > 18%
19. Watch bottom of EICAS for either A or B to be displayed next to the Valve state
20. Watch EICAS until N2 reaches > 55%
21. Watch bottom of EICAS for either A or B to disappear
22. Once Engine is started, turn Engine Mode Selector to NORM
23. Shut it down by turning off the corresponding Engine Master SW
24. Engine should shutdown and 5 minutes after the Engine Master SW was turned off the FADEC should unpower and the EICAS should show amber X
25. Repeat steps 16-18
26. This time the opposite Igniter should be used i.e B when it was A or vice versa 
27. Repeat for Engine 2

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_FADEC.js [hunk 1]
```diff
@@ -11,35 +12,47 @@ class A32NX_FADEC {
...
+        const engineState = SimVar.GetSimVarValue(`L:A32NX_ENGINE_STATE:${this.engine}`,"Number");
+        const n2Percent = SimVar.GetSimVarValue(`L:A32NX_ENGINE_N2:${this.engine}`,"Percent");
 
-        if ((this.dcEssPoweredInPreviousUpdate != dcEssIsPowered && dcEssIsPowered == 1) || this.lastMasterState != masterState) {
+        if ((this.dcEssPoweredInPreviousUpdate !== dcEssIsPowered && dcEssIsPowered === 1) ||
+            (this.lastEngineState !== engineState && engineState === 4) ) {
             this.fadecTimer = 5 * 60;
         }
+        if ((this.lastEngineState === 2 || this.lastEngineState === 3) && engineState !== 2 && engineState !== 3){
+            this.lastActiveIgniterAutostart ^= 1 // toggles Igniter
```
Comment (2hwk) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_FADEC.js:24]: Missing ;
```suggestion
            this.lastActiveIgniterAutostart ^= 1; // toggles Igniter
```

**Review by 2hwk — CHANGES_REQUESTED** [review]

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_FADEC.js [hunk 2]
```diff
@@ -11,35 +12,47 @@ class A32NX_FADEC {
...
+        }
+
+        this.igniting = ignitionState && (engineState === 2 || engineState === 3) && n2Percent > 18 && n2Percent < 55;
 
-        if (this.lastIgnitionState != ignitionState && !ignitionState) {
-            this.fadecTimer = 30;
+        if (this.lastIgnitionState !== ignitionState && !ignitionState) {
+            this.fadecTimer = Math.max(30,this.fadecTimer);
```
[the remark this answers is not in the document]
Comment (Maximilian-Reuter) [reply fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_FADEC.js:30]: js is weird, didn't realize semicolons were optional

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_FADEC.js [hunk 3]
```diff
@@ -3,6 +3,7 @@ class A32NX_FADEC {
         this.engine = engine;
         this.fadecTimer = -1;
         this.dcEssPoweredInPreviousUpdate = false;
+        this.lastActiveIgniterAutostart = 0 // 0 = A, 1 = B
```
Comment (Nufflee) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_FADEC.js:6]:
```suggestion
        this.lastActiveIgniterAutostart = 0; // 0 = A, 1 = B
```

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_FADEC.js [hunk 4]
```diff
@@ -11,35 +12,47 @@ class A32NX_FADEC {
 
     update(deltaTime) {
         const dcEssIsPowered = this.isDcEssPowered();
-        const masterState = SimVar.GetSimVarValue("FUELSYSTEM VALVE SWITCH:" + (this.engine), "Bool");
-        const ignitionState = SimVar.GetSimVarValue("L:XMLVAR_ENG_MODE_SEL", "Enum") == 2;
+        const ignitionState = SimVar.GetSimVarValue("L:XMLVAR_ENG_MODE_SEL", "Enum") === 2;
+        const engineState = SimVar.GetSimVarValue(`L:A32NX_ENGINE_STATE:${this.engine}`,"Number");
```
Comment (Nufflee) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_FADEC.js:16]:
```suggestion
        const engineState = SimVar.GetSimVarValue(`L:A32NX_ENGINE_STATE:${this.engine}`, "Number");
```

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_FADEC.js [hunk 5]
```diff
@@ -11,35 +12,47 @@ class A32NX_FADEC {
...
+        const n2Percent = SimVar.GetSimVarValue(`L:A32NX_ENGINE_N2:${this.engine}`,"Percent");
```
Comment (Nufflee) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_FADEC.js:17]:
```suggestion
        const n2Percent = SimVar.GetSimVarValue(`L:A32NX_ENGINE_N2:${this.engine}`, "Percent");
```

**Review by Nufflee — COMMENTED** [review]
wen js linting

**Comment by Maximilian-Reuter** [comment]
did I setup my VS Code incorrectly? Cause it never marked any of these things.

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #7919
Tier of Testing  : 1
Date             : 03/04/2023

Testing Process: 
Spawn C/D
Power Up A/C using ground power
Wait 5 minutes for initial FADEC power up period to elapse and EICAS to show amber X - **OK**
Use the FADEC GND PWR 1+2 Pushbuttons to activate the respective Fadecs and watch the EICAS show values - **OK**
Turn off pushbuttons and see amber X on EICAS - **OK**
Move Engine Mode Selector into CRANK
Watch EICAS display values instead of amber X - **OK**
Move Engine Mode Selector to NORM
Watch EICAS display amber X - **OK**
Move Engine Mode Selector to IGN/START
Watch EICAS display values instead of amber X - **OK**
Move Engine Mode Selector to NORM
Wait 30 seconds
Watch EICAS display amber X - **OK**
Turn on APU and APU BLEED and turn on Fuel Pumps
Move Engine Mode Selector to IGN/START
Move Engine Master SW to ON
Watch EICAS until N2 reaches > 18%
Watch bottom of EICAS for either A or B to be displayed next to the Valve state - **OK**
Watch EICAS until N2 reaches > 55%
Watch bottom of EICAS for either A or B to disappear - **OK**
Once Engine is started, turn Engine Mode Selector to NORM
Shut it down by turning off the corresponding Engine Master SW
Engine should shutdown and 5 minutes after the Engine Master SW was turned off the FADEC should unpower and the EICAS should show amber X - **OK**
Repeat steps 16-18
This time the opposite Igniter should be used i.e B when it was A or vice versa - **OK**
Repeat for Engine 2 - **OK**

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

**Review by 2hwk — APPROVED** [review]

**Pull request merged**